Bugfix invalid yara rules pass validation

cccs-gm opened the pull request:
Added a check that compiles the Yara Rule String prior to plyara parsing. To facilitate this yara-python>=4.0.2 was added to the requirements.txt file.

Also updated the MITRE ATT&CK sub-module to the latest master branch commit.

cccs-gm committed:
Added a new validation check

- uses the yara-python library to compile the rules file, if there are any errors return in a similar fashion to plyara parsing errors. This check is performed before sending
the file string contents to plyara for parsing
- added yara-python>=4.0.2 to the requirements file

cccs-gm committed:
Merge branch 'master' into bugfix_invalid_yara_rules_pass_validation

cccs-gm committed:
Bringing the MITRE ATT&CK submodule to current master commit

- As of Jan. 15th 2021

cccs-gm committed:
Bugfix of bug added by the original fix

- there were two other cases that caused rules files to be marked invalid when they should have been valid
  - any rules file that has an "includes" fails because the includes file cannot be found we now skip these to maintain the original behaviour
  - any rules files that were dependant on rules in a different file would also fail, we now ignore these errors

- rules that use modules but did not import the module are now marked as invalid
  - we added a flag to the cli '-m' to ignore this behaviour

cccs-gm committed:
Added the new flag to the README.md

cccs-jp commented on yara_validator_cli.py:
```diff
@@ -46,6 +46,9 @@
 parser.add_argument('-st', '--strict', action='store_true', default=False, dest='strict',
                     help='This causes the cli to return a non-zero exit code for warnings.')
+parser.add_argument('-m', '--module', action='store_false', default=True,
+                    dest='module', help='This flag disables the check that used modules in the rule file have been '
```
Need to be rewritten

cccs-jp commented on yara-validator/yara_validator.py:
```diff
@@ -92,13 +92,14 @@ def check_validator_cfg(validator_cfg):
         representation of the yara file to YaraValidator.valadation
     :param yara_file: The file variable passed in. Usually a string or Path variable
     :param generate_values: determine if the values the validator can generate should be generated or not, default True
+    :param check_import_modules: determines if there is a check that modules are imported in the rule, default True
```
Also need to be clarified

cccs-gm committed:
Upgraded descriptions

- made the description of the new flag clearer and more understandable
- made the comment about the new variable clearer and more understandable

cccs-gm commented:
Requested changes made to descriptions.
